### PR TITLE
feat(#110): object-does-not-match-filename lint

### LINKED ISSUE #110: every `.eo` file must have a single object (with the same name)
Creating a file `foo.eo` and then putting this code into it is possible (compiler won't complain):

```
[] > bar
```

However, it's confusing for the reader. Let's create a lint that will check that the file name is identical to the name of the object in it.

### PR DESCRIPTION
**Pull request by h1alexbel**
In this pull I've implemented new lint: `object-does-not-match-filename`, that issues `warning` defect in case top object name is different from the filename.

closes #110 
History:
- **feat(#110): bad/good test cases**
- **feat(#110): object-does-not-match-filename**
- **feat(#110): move**
- **feat(#110): passes**
- **feat(#110): stronger XPaths**
- **feat(#110): passes with children**
- **feat(#110): allows good name with chidren**
- **feat(#110): ignore in tests**
- **feat(#110): program name**

### CODE REVIEW

**Comment by h1alexbel** [comment]
@maxonfjvipon please check

File: src/main/resources/org/eolang/lints/names/object-does-not-match-filename.xsl [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+-->
+<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="2.0" id="object-does-not-match-filename">
+  <xsl:output encoding="UTF-8" method="xml"/>
+  <xsl:variable name="filename" select="/program/@name"/>
+  <xsl:variable name="tested" select="/program/metas/meta[head='tests']"/>
+  <xsl:template match="/">
+    <defects>
+      <xsl:apply-templates select="/program/objects/o[@name != $filename]" mode="with-data"/>
```
Comment (maxonfjvipon) [inline src/main/resources/org/eolang/lints/names/object-does-not-match-filename.xsl:31]: @h1alexbel I think mode `with-data` is confusing here, let's add some meaningful name to the mode

**Review by maxonfjvipon — CHANGES_REQUESTED** [review]
@h1alexbel looks good, just one small change is required

**Comment by h1alexbel** [comment]
@maxonfjvipon updated. Take a look, please

**Review by maxonfjvipon — APPROVED** [review]
@h1alexbel looks good now, thanks!

**Comment by h1alexbel** [comment]
@yegor256 take a look, please

**Review by yegor256 — APPROVED** [review]
@h1alexbel LGTM

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**